### DISCUSSION TITLE
How to use EM_ASM in .c file?

### DISCUSSION THREAD

**Discussion by s2415372** [emscripten-core/emscripten#16775] [Q&A] [answered]
Hello, I'm newbie of emscripten.
I follow the example on website to do practice.
But I can't use EM_ASM in my code.
The following is my code.
[provide.zip](https://github.com/emscripten-core/emscripten/files/8532484/provide.zip)

When I enable EM_ASM,
functions.wasm will appear

> (import "env" "emscripten_asm_const_int" (func $env.emscripten_asm_const_int (type $t4)))

I tried copy the code from functions.js
But will get error.

> script.js:41 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '5243939')
>     at readAsmConstArgs (script.js:41:23)
>     at _emscripten_asm_const_int (script.js:53:16)
>     at functions.wasm:0x212
>     at script.js:73:25

emsdk version : 3.1.8
python version : 3.10
build-cmd : emcc ./functions.c -s WASM=1 -o ./functions.html

Any help with this will be appreciated. 
I don't know what else to do or to change. 
Thank you.

---
>Leave my final simple code here, maybe others people will need.
[final.zip](https://github.com/emscripten-core/emscripten/files/8533525/final.zip)

**Comment by sbc100** [comment, marked as answer]
It sounds like maybe `HEAPU8` is not defined.. which would be very odd... did you include the entire functions.js file?  How did you load the JS generated by emscripten?

**Reply by s2415372** [reply]
Thank you so much.
The articles I saw before didn't said should include entire .js
After include .js , it works now.
